### PR TITLE
Test API headers can be included standalone. 

### PR DESCRIPTION
**Pull request by hsorby**
This pull request adds generated tests that check that each API header can be included on its own.

Addresses #178.

### CODE REVIEW

**Comment by hsorby** [comment]
Looks like I haven't made these new tests smart enough to run on Windows 😒.

**Comment by hsorby** [comment]
Ready for review.

File: tests/api_headers/api_header_test.in.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (agarny) [inline tests/api_headers/api_header_test.in.cmake:1]: Why that empty line?

File: tests/api_headers/api_header_test.in.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  FOLDER tests)
+
+add_test(NAME api_header_inclusion_${CURRENT_TEST} COMMAND ${CURRENT_TEST})
+
+if(MSVC)
+  if(HAVE_LIBXML2_CONFIG)
+	set_tests_properties(api_header_inclusion_${CURRENT_TEST}
+	PROPERTIES ENVIRONMENT "PATH=$<TARGET_FILE_DIR:cellml>\;$<TARGET_FILE_DIR:gtest_main>\;$<TARGET_FILE_DIR:xml2>")
+  else()
+	# The libxml2 dll must be on the path otherwise the tests will not be able to run.
```
Comment (agarny) [inline tests/api_headers/api_header_test.in.cmake:17]: Pedantic: `DLL` not `dll`.

**Review by agarny — APPROVED** [review]
Looks good to me.

**Review by nickerso — APPROVED** [review]

**Pull request merged**